### PR TITLE
Revert Handle stuck queued tasks in Celery for db backend

### PR DESCRIPTION
**Pull request by ephraimbuddy**
This reverts #19769 and #21335

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/executors/celery_executor.py [hunk 1]
```diff
@@ -398,28 +403,34 @@ def _clear_stuck_queued_tasks(self, session: Session = NEW_SESSION) -> None:
...
-            ]
         self.log.debug("Checking for stuck queued tasks")
 
         max_allowed_time = utcnow() - self.task_adoption_timeout
 
-        for task in session.query(TaskInstance).filter(
-            TaskInstance.state == State.QUEUED, TaskInstance.queued_dttm < max_allowed_time
-        ):
-            if task.key in self.queued_tasks or task.key in self.running:
-                continue
```
Comment (jedcunningham) [inline airflow/executors/celery_executor.py:425]: Why don't we want this short-circuit any longer?

File: airflow/executors/celery_executor.py [hunk 2]
```diff
@@ -342,7 +342,12 @@ def sync(self) -> None:
         if self.adopted_task_timeouts:
             self._check_for_stalled_adopted_tasks()
         if time.time() - self.stuck_tasks_last_check_time > self.stuck_queued_task_check_interval:
-            self._clear_stuck_queued_tasks()
+            try:
+                with timeout(seconds=OPERATION_TIMEOUT):
```
Comment (jedcunningham) [inline airflow/executors/celery_executor.py]: I'm a little torn on this. Should we just guard the celery query? I'm also not sure we should be silent about it.

The default of 1s short for this whole operation too imo.

I'm not sure we should have this at all, being db backed?

File: airflow/executors/celery_executor.py (hunk 1 above)
Comment (ephraimbuddy) [reply airflow/executors/celery_executor.py:425]: From this comment https://github.com/apache/airflow/issues/21225#issuecomment-1031543120 and the comments at https://github.com/apache/airflow/pull/19769, the tasks are added back to celery by adoption.

File: airflow/executors/celery_executor.py (hunk 2 above)
Comment (ephraimbuddy) [reply airflow/executors/celery_executor.py]: I think this check is not so important and timeout error from here especially as we make connection to celery(which is a network request) should be ignored. The 1s is low. I will move the timeout to celery db query

File: airflow/executors/celery_executor.py [hunk 3]
```diff
@@ -402,28 +404,39 @@ def _clear_stuck_queued_tasks(self, session: Session = NEW_SESSION) -> None:
...
+                    with timeout(seconds=15):
+                        celery_task_ids = [
+                            t.task_id
+                            for t in session_.query(task_cls.task_id)
+                            .filter(~task_cls.status.in_([celery_states.SUCCESS, celery_states.FAILURE]))
+                            .filter(task_cls.task_id.in_([ti.external_executor_id for ti in queued_too_log]))
+                            .all()
+                        ]
+                except Exception:
+                    # This timeout is not important, we should skip
```
Comment (Chris7) [inline airflow/executors/celery_executor.py]: If this exception is hit, `celery_task_ids` will not be set, and there will be `NameError` from `if task.external_executor_id in celery_task_ids`

**Comment by ephraimbuddy** [comment]
Reverting the previous PRs as no one is offering to test and I can't reproduce this locally

**Review by jedcunningham — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**